engine/access/ingestion/engine_test.go

```diff
@@ -131,6 +131,7 @@ func (s *Suite) SetupTest() {
 	s.receipts = new(storagemock.ExecutionReceipts)
 	s.transactions = new(storagemock.Transactions)
 	s.results = new(storagemock.ExecutionResults)
+	s.results.On("BatchIndex", mock.Anything, mock.Anything, mock.Anything, mock.Anything).Return(nil).Maybe()
```

Comment:
would be great to verify that the lock `storage.LockIndexExecutionResult` is held